src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,879 @@ const ActorInit Object_Kankyo_InitVars = {
+            }
+            if (Flags_GetEventChkInf(0xAD)) {
+                this->effects[5].size = 0.0f;
+            }
+
+            if (gSaveContext.cutsceneTrigger != 0) {
+                if (gSaveContext.entranceIndex == 0x0538) {
+                    this->effects[0].size = 0.1f;
+                }
+                if (gSaveContext.entranceIndex == 0x053C) {
+                    this->effects[1].size = 0.1f;
+                }
+                if (gSaveContext.entranceIndex == 0x0540) {
+                    this->effects[2].size = 0.1f;
+                }
+                if (gSaveContext.entranceIndex == 0x0544) {
+                    this->effects[3].size = 0.1f;
+                }
+                if (gSaveContext.entranceIndex == 0x0548) {
+                    this->effects[4].size = 0.1f;
+                }
+                if (gSaveContext.entranceIndex == 0x054C) {
+                    this->effects[5].size = 0.1f;
+                }
+            }
+
+            this->demoObjectLoaded = false;
+            ObjectKankyo_SetupAction(this, ObjectKankyo_InitBeams);
+            break;
+    }
+}
+
+void ObjectKankyo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Actor_Kill(thisx);
+}
+
+void ObjectKankyo_Snow(ObjectKankyo* this, GlobalContext* globalCtx) {
+}
+
+void ObjectKankyo_Fairies(ObjectKankyo* this, GlobalContext* globalCtx) {
+    static Vec3f sSoundPos = { 0.0f, 0.0f, 0.0f };
+    Player* player;
+    f32 dist;
+    s32 playerMoved;
+    f32 dx;
+    f32 dy;
+    f32 dz;
+    f32 offsetX;
+    f32 offsetY;
+    f32 offsetZ;
+    f32 maxDist;
+    f32 baseX;
+    f32 baseY;
+    f32 baseZ;
+    Vec3f vec1 = { 0.0f, 0.0f, 0.0f };
+    Vec3f vec2 = { 0.0f, 0.0f, 0.0f };
+    f32 random;
+    s16 i;
+    Vec3f offset;
+
+    player = GET_PLAYER(globalCtx);
+
+    if (globalCtx->sceneNum == SCENE_SPOT04 && gSaveContext.sceneSetupIndex == 7) {
+        dist = Math3D_Vec3f_DistXYZ(&this->lastEyePos, &globalCtx->view.eye);
+
+        this->lastEyePos.x = globalCtx->view.eye.x;
+        this->lastEyePos.y = globalCtx->view.eye.y;
+        this->lastEyePos.z = globalCtx->view.eye.z;
+
+        dist /= 30.0f;
+        if (dist > 1.0f) {
+            dist = 1.0f;
+        }
+
+        func_800F436C(&sSoundPos, NA_SE_EV_NAVY_FLY - SFX_FLAG, (0.4f * dist) + 0.6f);
+        switch (globalCtx->csCtx.frames) {
+            case 473:
+                func_800788CC(NA_SE_VO_NA_HELLO_3);
+                break;
+
+            case 583:
+                func_800F4524(&D_801333D4, NA_SE_VO_NA_HELLO_2, 32);
+                break;
+
+            case 763:
+                func_80078884(NA_SE_EV_NAVY_CRASH - SFX_FLAG);
+                break;
+
+            case 771:
+                func_80078884(NA_SE_VO_RT_THROW);
+                break;
+        }
+    }
+
+    if (globalCtx->envCtx.unk_EE[3] < 64 &&
+        (gSaveContext.entranceIndex != 0x00EE || gSaveContext.sceneSetupIndex != 4 || globalCtx->envCtx.unk_EE[3])) {
```

Comment:
both != 0 and demorgan's law break matching

Reply:
```suggestion
        !(gSaveContext.entranceIndex == 0x00EE && gSaveContext.sceneSetupIndex == 4 && !globalCtx->envCtx.unk_EE[3])) {
```
does match, idk if it's much better though because of `!globalCtx->envCtx.unk_EE[3]`
as you want

(I like the `!(... && )` better because it makes it more obvious those conditions are for excluding a specific something, imho)